File: Kol_2/2024_Januar/client/models/automobil.js
import { iznajmiAuto } from "../utils.js";

export class Automobil {
  constructor(
    automobilID,
    model,
    predjenaKilometraza,
    godiste,
    cenaPoDanu,
    iznajmljen
  ) {
    this.automobilID = automobilID;
    this.model = model;
    this.predjenaKilometraza = predjenaKilometraza;
    this.godiste = godiste;
    this.cenaPoDanu = cenaPoDanu;
    this.iznajmljen = iznajmljen;
    this.autoDiv;
  }
}

File: Kol_2/2024_Jun_A/server/Models/Automobil.cs
using System.ComponentModel.DataAnnotations;
using System.Text.Json.Serialization;

namespace rent_a_car.Models
{
    public class Automobil
    {
        [Key]
        public int AutomobilID { get; set; }

        public required string Model { get; set; }

        public required int PredjenaKilometraza { get; set; }

        public required int Godiste { get; set; }

        public required int CenaPoDanu { get; set; }

        public required bool Iznajmljen { get; set; } = false;

        public List<Iznajmljivanje>? UgovorIznajmljivanja { get; set; }

        public Automobil()
        {
            UgovorIznajmljivanja = [];
        }
    }
}